### PR TITLE
Feat: Extract and adds tokens to Dropdown

### PR DESCRIPTION
**Pull request by hellofanny**
## What's the purpose of this pull request?

- Extracts `Dropdown` component from `Breadcrumb`
- Applies the new https://github.com/vtex-sites/base.store/pull/407 to `Dropdown`: new scoped tokens linked to global ones.
- Adds `size` prop to `DropdownMenu`

## How does it work?

Some visual changes were made to the Dropdown on `Breadcrumb`.
|[Before](https://nextjs-store-storybook-faststore.vercel.app/?path=/docs/molecules-breadcrumb--default)|[Now](https://nextjs-store-storybook-git-feat-theming-dropdo-374add-faststore.vercel.app/?path=/docs/molecules-breadcrumb--default)|
|-|-|
before|now

- Icon default color is blue;
- Added missing light gray divider between items;

Tokens applied only for `DropdownMenu` and `DropdownItem`, `DropdownButton` will be revisit soon and doesn't have a default style.

## How to test it?

- The dropdown should be working fine on the `Breadcrumb` as before these changes, but with the previously mentioned visual changes.
- Check the `Dropdown` documentation in the [preview]( https://nextjs-store-storybook-git-feat-theming-dropdo-374add-faststore.vercel.app/?path=/docs/molecules-dropdown-dropdownmenu--default-story) 

## Checklist

<em>You may erase this after checking them all ;)</em>

**Changelog**
- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#4](https://github.com/vtex-sites/nextjs.store/pull/4))* 

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.

**Documentation**
- [x] PR description
- [x] Added to/Updated the Storybook - *if applicable*.
- [ ] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update.

### CODE REVIEW

**Review by renatamottam — APPROVED** [review]
Looks amazing 😍

**Review by filipewl — COMMENTED** [review]
Looking awersome! ✨

File: src/components/ui/Dropdown/dropdown.module.scss [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  --fs-dropdown-item-small-text-size        : var(--fs-text-size-1);
+
+  // --------------------------------------------------------
+  // Structural Styles
+  // --------------------------------------------------------
+
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  max-width: 50%;
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/ui/Dropdown/dropdown.module.scss:43]: Maybe we can remove it and apply it only for specific cases. What do you think?

File: src/components/ui/Dropdown/Dropdown.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+      <p>An item of a set of items on <code>DropdownMenu</code>.</p>
+    </article>
+  </article>
+</section>
+
+---
+
+## Accessibility
+
+### Keyboard Interactions
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/ui/Dropdown/Dropdown.stories.mdx:165]: 💯

File: src/components/ui/Dropdown/Dropdown.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+      <h3>Dropdown Menu</h3>
+      <p>
+        Displays a set of actions/items to the user, usually used to show a menu of options.
+      </p>
+    </article>
+
+  </article>
+  <article>
+    <div style={{ height: '225px' }}>
+      <a href="?path=/docs/molecules-dropdown-dropdownitem--default-story"><img src="https://user-images.githubusercontent.com/3356699/173459414-f03c0d5b-7320-4cf0-9cc4-67ac37d234f8.png"/></a>
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/ui/Dropdown/Dropdown.stories.mdx:152]: Niceee! I will fix it ~

File: src/components/ui/Dropdown/Dropdown.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+import {
+  TokenTable,
+  TokenRow,
+  TokenDivider,
+} from 'src/../.storybook/components'
+
+<Meta title="Molecules/Dropdown/Default" />
+
+export const Template = ({ id, ...args }) => (
```
Comment (eduardoformiga) [inline src/components/ui/Dropdown/Dropdown.stories.mdx:16]: I think we are using only args for all templates
```suggestion
export const Template = (args) => (
```

File: src/components/ui/Dropdown/Dropdown.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    </DropdownButton>
+    <DropdownMenu {...args}>
+      <DropdownItem>Dropdown Item 1</DropdownItem>
+      <DropdownItem>Dropdown Item 2</DropdownItem>
+      <DropdownItem>Dropdown Item 3</DropdownItem>
+    </DropdownMenu>
+  </Dropdown>
+)
+
+export const TemplateWithIcon = ({ id, ...args }) => (
```
Comment (eduardoformiga) [inline src/components/ui/Dropdown/Dropdown.stories.mdx:33]: the same thing here and for the others

File: src/components/ui/Dropdown/Dropdown.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+</Canvas>
+
+---
+
+## Related Components
+
+<section className="sbdocs-section">
+  <article>
+    <div style={{ height: '225px' }}>
+      <a href="?path=/docs/molecules-dropdown-dropdownmenu--default-story"><img src="https://user-images.githubusercontent.com/3356699/173459468-1a18c582-85bb-4d27-a30a-2a4ee79bad2c.png"/></a>
```
[the remark this answers is not in the document]
Comment (eduardoformiga) [reply src/components/ui/Dropdown/Dropdown.stories.mdx:140]: hmm, maybe using https://vtexhelp.vtexassets.com/ ?

**Review by eduardoformiga — COMMENTED** [review]
Amazing Job! ✨ 

the dropdown menu should keep the dropdown item width? or should we change it to 100%?

Screen Shot 2022-06-14 at 15 52 22

File: src/components/ui/Dropdown/dropdown.module.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/ui/Dropdown/dropdown.module.scss]: yeah! I've changed at  [000110f](https://github.com/vtex-sites/nextjs.store/pull/118/commits/000110fad65bb0e089d734693e583dc8af1e0649) 🥳

File: src/components/ui/Dropdown/DropdownButton.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+function DropdownButton({
+  children,
+  testId = 'store-dropdown-button',
+  ...otherProps
+}: DropdownButtonProps) {
+  return (
+    <UIDropdownButton
+      data-fs-dropdown-button
+      className={styles.fsDropdownButton}
```
[the remark this answers is not in the document]
Comment (hellofanny) [reply src/components/ui/Dropdown/DropdownButton.tsx]: Removed at [c0fed08](https://github.com/vtex-sites/nextjs.store/pull/118/commits/c0fed08fa2fa6ddcd99d015b28ac67e4531b17c5) ✅

**Review by eduardoformiga — APPROVED** [review]

File: src/components/ui/Dropdown/Dropdown.stories.mdx (hunk 4 above)
Comment (hellofanny) [reply src/components/ui/Dropdown/Dropdown.stories.mdx]: [done](https://github.com/vtex-sites/nextjs.store/pull/118/commits/62ad8548f49725fa9e7332e237e277078240f10b)! 🥳  Thanks!

File: src/components/ui/Dropdown/Dropdown.stories.mdx (hunk 5 above)
Comment (hellofanny) [reply src/components/ui/Dropdown/Dropdown.stories.mdx]: [done](https://github.com/vtex-sites/nextjs.store/pull/118/commits/62ad8548f49725fa9e7332e237e277078240f10b)! 🥳

**Comment by hellofanny** [comment]
> Amazing Job! ✨
> 
> the dropdown menu should keep the dropdown item width? or should we change it to 100%?
> 
> Screen Shot 2022-06-14 at 15 52 22

Thanks for reviewing ✨ . I've checked it with the designers; we are going to keep it like that for now, ok?!

**Review by filipewl — APPROVED** [review]

**Pull request merged**